src/Result/src/ResultFactory.cpp

```diff
@@ -20,15 +22,15 @@ namespace calculator
 // ResultFactory public methods /
 ///
    
-    Result ResultFactory::createResult(const Expression &parsedExpression, float answer)
+    std::shared_ptr<IResult> ResultFactory::createResult(const Expression &parsedExpression, float answer)
     {
         std::stringstream stringStream;
         
         stringStream << parsedExpression.a << CalculatorMessages::EMPTY_SPACE 
         << parsedExpression.operation << CalculatorMessages::EMPTY_SPACE 
         << parsedExpression.b << CalculatorMessages::EQUALS << answer << std::endl;
         
-        return Result(parsedExpression, answer, stringStream.str());
+        return std::shared_ptr<IResult> (new Result(parsedExpression, answer, stringStream.str()));
```

Comment:
You don't need the new keyword, by creating the pointer it should be enough and passing the arguments. 
std::shared_ptr<IResult> (parsedExpression, answer, stringStream.str());